### DISCUSSION TITLE
v0.8.1

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#284] [Announcements]
**新功能**
1. 支持自定义裁剪 #268 
2. 重新添加俄语翻译（感谢 @0x4E69676874466F78 ）
3. 现在托盘图标的文字提示包含当前配置信息

**修复**
1. 降低了 Graphics Capture 捕获模式的 CPU 占用
2. 优化了 Desktop Duplication 捕获模式的流畅度
3. 修复了特定情况下光标无法在屏幕间移动的问题  #270 
4. 优化了 CRT_Easymode 效果的参数

<hr /><em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/v0.8.1'>v0.8.1</a>.</em>

**Comment by chaos700** [comment]
现在的自定义裁剪功能有点难用，
要在设定里填充上下左右的像素，本身就很难直观知道要裁剪的像素，
而且不同程序要裁剪的部分未必相同，意味着每次都要手动调整参数，
能不能像Windows自带截图工具里的矩形截图模式，可以用鼠标手动拉出矩形区域，指定那个区域来放大画面。

**Reply by Blinue** [reply]
设计这个功能的时候我假设它是很少用到的 https://github.com/Blinue/Magpie/issues/268#issuecomment-1008255780 ，需要自定义裁剪的窗口应该很少，场景也比较简单

如果以后可以保存针对特定游戏的配置，操作会简化很多

**Comment by NoRunRad** [comment]
Hello
I use an online translator.
FSR 2.0 will be announced soon. In connection with this event, the question arose: will Magpie be released on this technology?

**Reply by Blinue** [reply]
I can't give a definitive answer until they release the source code, but I'll do my best.

**Reply by bloc97** [reply]
@NoRunRad FSR 2.0 uses motion vectors from the game engine to implement TAA + super-resolution, which are not available to Magpie. However RSR could be implemented, but it might be the same code as FSR.

**Reply by Blinue** [reply]
bloc97 is right. According to [this blog](https://gpuopen.com/fsr2-announce/), FSR 2.0 is designed to be integrated in games only, it is not a generic post-processing algorithm. (which will definitely slow down its rollout)

![image](https://user-images.githubusercontent.com/34770031/159386820-4e14d895-86cd-44f5-aac1-0013a96cc7b0.png)

**Reply by NoRunRad** [reply]
Did I understand correctly that FSR 2.0 will only be available in the game if the developer adds it there?
Perhaps it will be possible to add FSR 2.0 files to the game so that it becomes possible to use the technology?Maybe iIt is possible through a software shell like Experience or something like that.

**Reply by Blinue** [reply]
I don't know if there is a generic way to "inject" FSR into games, it's not how Magpie works. FSR 2.0 can't even be implemented at the driver level, you have to wait for game developers to integrate it.

**Reply by NoRunRad** [reply]
Very sad(
But in any case, thanks for the Magpie.

**Reply by finestuff** [reply]
Judging by the latest mods it became possible to simply replace DLSS with FSR2.

**Comment by NoRunRad** [comment]
Hello.
Is it possible to make edits to the algorithm?
 When I play Diablo 2 and stretch the image from 800x600, the colors change a lot. I understand why this is happening, but is it possible to change the algorithm so that it tries to finish the colors closer to the original ones?Some inscriptions become obscure.For example: the yellow and orange labels become almost identical in color.It remains only to guess by the name of the object about its true value))

**Reply by Blinue** [reply]
> Is it possible to make edits to the algorithm?

You can modify the files in the "effects" folder. For FSR you need to modify FSR_EASU.hlsl and FSR_RCAS.hlsl.

**Reply by NoRunRad** [reply]
I can`t do it)I don`t know how)
But if it possible it is good.Maybe I can understand how do this but I doubt.

**Reply by Tzugimaa** [reply]
Clamp can limit color range to preserve original color, but may increase blurriness sir.

**Reply by NoRunRad** [reply]
It is Diablo II)Blurriness is not a problem)

**Reply by Tzugimaa** [reply]
Have you tried Jinc or Catmull-Rom?

**Comment by zahuifan** [comment]
真诚感谢各位技术大佬的分享！据说FSR2.0已发布，请问Magpie能否支持？

**Reply by hooke007** [reply]
https://github.com/Blinue/Magpie/discussions/284#discussioncomment-2400536

**Reply by greatyingzi** [reply]
> 

上边不是已经回复了吗？fsr2.0可能要游戏内部集成才行，[https://github.com/Blinue/Magpie/discussions/284#discussioncomment-2407611](url)如果真的如此，那大概率是没办法支持的。
